### ISSUE TITLE
[BUG] plotting.plot_img_on_surf gives an error if alpha is set low

### ISSUE THREAD

**Issue by paulcbogdan** [nilearn/nilearn#4900]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Operating system

- [ ] Linux
- [ ] Mac
- [x] Windows

### Operating system version

Windows 11

### Python version

- [ ] 3.13
- [x] 3.12
- [ ] 3.11
- [ ] 3.10
- [ ] 3.9

### nilearn version

Version 0.11.0 (I just updated for this)

### Expected behavior

 `plotting.plot_img_on_surf` is supposed to make an image, but it instead errors.

### Current behavior & error messages

I was using `plotting.plot_img_on_surf`, and it produced an error when I specified the `threshold` parameter and I set `alpha` to a low value. For example:

```Python
from nilearn import datasets, plotting, image
data = datasets.fetch_abide_pcp(n_subjects=1)
fig, axs = plotting.plot_img_on_surf(
    image.index_img(data['func_preproc'][0], 0),
    threshold=2, alpha=.1, )
plt.show()
```

Produces the error:

```
  File "...\", line ..., in <module>
    fig, axs = plotting.plot_img_on_surf(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\Lib\site-packages\nilearn\plotting\surf_plotting.py", line 1841, in plot_img_on_surf
    plot_surf_stat_map(
  File "...\Lib\site-packages\nilearn\plotting\surf_plotting.py", line 1517, in plot_surf_stat_map
    display = plot_surf(
              ^^^^^^^^^^
  File "...\Lib\site-packages\nilearn\plotting\surf_plotting.py", line 989, in plot_surf
    fig = _plot_surf_matplotlib(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "...\Lib\site-packages\nilearn\plotting\surf_plotting.py", line 722, in _plot_surf_matplotlib
    p3dcollec.set_facecolors(face_colors)
  File "...\Lib\site-packages\matplotlib\_api\__init__.py", line 247, in method
    return getattr(self, name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\Lib\site-packages\mpl_toolkits\mplot3d\art3d.py", line 1077, in set_facecolor
    super().set_facecolor(colors)
  File "...\Lib\site-packages\matplotlib\collections.py", line 784, in set_facecolor
    self._set_facecolor(c)
  File "...\Lib\site-packages\matplotlib\collections.py", line 766, in _set_facecolor
    self._facecolors = mcolors.to_rgba_array(c, self._alpha)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\Lib\site-packages\matplotlib\colors.py", line 476, in to_rgba_array
    raise ValueError("RGBA values should be within 0-1 range")
ValueError: RGBA values should be within 0-1 range
```

This error doesn't occur if threshold is undefined or `alpha` is kept as some relatively high value (e.g., over 0.5). Interestingly, when preparing this example, I also tried passing MNI152_FILE_PATH (a different dataset) to `plot_img_on_surf`, which does not produce any error (instead, it just shows the image, although I know an anatomical image won't really be meaningful for a surface plot).

It seems that the `face_colors` variable in `plotting.surf_plotting._plot_surf_matplotlib` sometimes surpasses 1.0 due to floating point stuff (1.0000000000000002), which causes this.

### Steps and code to reproduce bug

I provided an error message above. I have a simple fix for this floating point stuff. To `plotting.surf_plotting._plot_surf_matplotlib`, I have simply added one line: `face_colors[face_colors > 1] = 1.` (see line 722 in this picture: https://imgur.com/PKkNUvj)

I have a pull request ready with this one-line addition, but I figured I should post this issue here first.

**Comment by Remi-Gau** [comment]
quick check bug existed as far back as 0.10.1: testing before was just too annoying for now

**Comment by paulcbogdan** [comment]
Thanks for confirming this, should I make a PR?

**Comment by Remi-Gau** [comment]
yeah open a PR and we can discuss it further there

make sure to add a regression test to 'demonstrate' the bug first on main and then fix it, so we make sure it does not come back

**Comment by paulcbogdan** [comment]
Okay! On a related note and because this seems like an actively developed feature in NiLearn, my feeling is that tiny alpha values like I tested are necessary if you actually want to make the mesh transparent. I'm guessing that because the mesh folds over itself. For example, I couldn't visually see any transparency when I set it to like 0.5. I needed to set it alpha = 0.03 to get this: https://imgur.com/a/n5XUiCA (the transparency does look very nice at this alpha)
